consensus/istanbul/backend/engine.go

```diff
@@ -234,17 +234,17 @@ func (sb *backend) verifyHeader(chain consensus.ChainReader, header *types.Heade
 		return errInvalidBlockScore
 	}
 
-	// TODO-kaiax: further flatten the code inside
-	if err := sb.verifyCascadingFields(chain, header, parents); err != nil {
-		return err
-	}
-
 	for _, module := range sb.consensusModules {
 		if err := module.VerifyHeader(header); err != nil {
 			return err
 		}
 	}
 
+	// TODO-kaiax: further flatten the code inside
+	if err := sb.verifyCascadingFields(chain, header, parents); err != nil {
+		return err
+	}
+
```

Comment:
If your intention was to invoke `module.VerifyHeader()` unconditionally, how about adding `defer` or some other formal rule to be applied in other sites? If this is not the one you pointed out, please let me know what you claimed.

Reply:
It's discussed in the scrum, but I'll leave it for history: it's ok not to be called on error, but I wanted to ignore `errEmptyCommittedSeals`. Maybe this can be further refactored when we flatten this part.